mlflow/azureml/__init__.py

```diff
@@ -380,15 +382,45 @@ def deploy(
             _copy_file_or_tree(src=absolute_model_path, dst=model_directory_path),
         )
 
-        registered_model = AzureModel.register(
-            workspace=workspace, model_path=tmp_model_path, model_name=model_name, tags=tags
-        )
+        registered_model = None
+        m_id = None
 
-        _logger.info(
-            "Registered an Azure Model with name: `%s` and version: `%s`",
-            registered_model.name,
-            registered_model.version,
-        )
+        if model_uri.startswith("models:/"):
+            m_name = model_uri.split("/")[-2]
+            m_version = int(model_uri.split("/")[-1])
+            m_id = "{}:{}".format(m_name, m_version)
```

Comment:
Is there a way to go from /production to /version @dbczumar ? We might be able to expand that out to avoid the issue you brought up